src/MOI_wrapper.jl

```diff
@@ -1,58 +1,99 @@
-using LinQuadOptInterface
-
-const LQOI = LinQuadOptInterface
-const MOI  = LQOI.MOI
-
-const SUPPORTED_OBJECTIVES = [
-    LQOI.SinVar,
-    LQOI.Linear,
-    LQOI.Quad
-]
+import MathOptInterface
+
+const MOI = MathOptInterface
+const CleverDicts = MOI.Utilities.CleverDicts
+
+@enum(VariableType, CONTINUOUS, BINARY, INTEGER, SEMIINTEGER, SEMICONTINUOUS)
+@enum(BoundType, NONE, LESS_THAN, GREATER_THAN, LESS_AND_GREATER_THAN, INTERVAL, EQUAL_TO)
+@enum(ObjectiveType, SINGLE_VARIABLE, SCALAR_AFFINE, SCALAR_QUADRATIC)
+
+mutable struct VariableInfo
+    index::MOI.VariableIndex
+    column::Int
+    bound::BoundType
+    type::VariableType
+    start::Union{Float64, Nothing}
+    name::String
+    # Storage for constraint names associated with variables because Gurobi
+    # can only store names for variables and proper constraints.
+    # We can perform an optimization and only store three strings for the
+    # constraint names because, at most, there can be three SingleVariable
+    # constraints, e.g., LessThan, GreaterThan, and Integer.
+    lessthan_name::String
+    greaterthan_interval_or_equalto_name::String
+    type_constraint_name::String
+    function VariableInfo(index::MOI.VariableIndex, column::Int)
+        return new(index, column, NONE, CONTINUOUS, nothing, "", "", "", "")
+    end
+end
 
-const SUPPORTED_CONSTRAINTS = [
-    (LQOI.Linear, LQOI.EQ),
-    (LQOI.Linear, LQOI.LE),
-    (LQOI.Linear, LQOI.GE),
-    # (Linear, IV),
-    (LQOI.Quad, LQOI.EQ),
-    (LQOI.Quad, LQOI.LE),
-    (LQOI.Quad, LQOI.GE),
-    (LQOI.SinVar, LQOI.EQ),
-    (LQOI.SinVar, LQOI.LE),
-    (LQOI.SinVar, LQOI.GE),
-    (LQOI.SinVar, LQOI.IV),
-    (LQOI.SinVar, MOI.ZeroOne),
-    (LQOI.SinVar, MOI.Integer),
-    (LQOI.VecVar, LQOI.SOS1),
-    (LQOI.VecVar, LQOI.SOS2),
-    (LQOI.SinVar, MOI.Semicontinuous{Float64}),
-    (LQOI.SinVar, MOI.Semiinteger{Float64}),
-    (LQOI.VecVar, MOI.Nonnegatives),
-    (LQOI.VecVar, MOI.Nonpositives),
-    (LQOI.VecVar, MOI.Zeros),
-    (LQOI.VecLin, MOI.Nonnegatives),
-    (LQOI.VecLin, MOI.Nonpositives),
-    (LQOI.VecLin, MOI.Zeros)
-]
+mutable struct ConstraintInfo
+    row::Int
+    set::MOI.AbstractSet
+    # Storage for constraint names. Where possible, these are also stored in the
+    # Gurobi model.
+    name::String
+    ConstraintInfo(row::Int, set) = new(row, set, "")
+end
 
-mutable struct Optimizer <: LQOI.LinQuadOptimizer
-    LQOI.@LinQuadOptimizerBase(Model)
+mutable struct Optimizer <: MOI.ModelLike
```

Comment:
Shouldn't `Optimizer` be a child of `MOI.AbstractOptimizer`, since it does expose `MOI.optimize!`?